### PR TITLE
[network, test-only] Network testing races

### PR DESCRIPTION
**Pull request by huitseeker**
Removes a potential few data races in network tests

### CODE REVIEW

File: network/test/echoengine_test.go [hunk 1]
```diff
@@ -673,7 +695,7 @@ func (suite *EchoEngineTestSuite) skipTest(reason string) {
 // assertMessageReceived asserts that the given message was received on the given channel
 // for the given engine
 func assertMessageReceived(t *testing.T, e *EchoEngine, m *message.TestMessage, c network.Channel) {
-	// wrap blocking channel reads with a timeout
+	// wrap bRLocking channel reads with a timeout
```
Comment (smnzhu) [inline network/test/echoengine_test.go:698]: typo?

File: network/p2p/pubsub_test.go [hunk 2]
```diff
@@ -42,10 +43,13 @@ func (suite *PubSubTestSuite) SetupTest() {
...
 
 type mockDiscovery struct {
-	peers []peer.AddrInfo
+	peerLock sync.Mutex
+	peers    []peer.AddrInfo
 }
 
 func (s *mockDiscovery) SetPeers(peers []peer.AddrInfo) {
+	defer s.peerLock.Unlock()
+	s.peerLock.Lock()
```
Comment (smnzhu) [inline network/p2p/pubsub_test.go:52]: nit: probably doesn't matter here, but in general I think it's best practice to reverse the order of these two lines: https://stackoverflow.com/a/51725166

File: network/queue/queueWorker_test.go [hunk 3]
```diff
@@ -41,17 +42,20 @@ func testWorkers(t *testing.T, maxPriority int, messageCnt int, workerCnt int) {
...
 	// callback checks if message is of expected priority
 	callback := func(data interface{}) {
 		actual := data.(int)
+		l.Lock()
 		assert.LessOrEqual(t, expectedPriority, actual)
 		atomic.AddInt64(&callbackCnt, 1)
 		if callbackCnt%int64(messagesPerPriority) == 0 {
 			expectedPriority--
 		}
+		l.Unlock()
```
Comment (smnzhu) [inline network/queue/queueWorker_test.go:58]: 👍

**Review by smnzhu — APPROVED** [review]

**Review by vishalchangrani — APPROVED** [review]
lgtm

**Pull request merged**